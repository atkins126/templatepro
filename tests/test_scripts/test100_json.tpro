** JSON **
{{loop(jsonobj.people) as person}}
{{:person.@@index}}) {{:person.first_name|uppercase}} {{:person.last_name|uppercase}} (age {{:person.age}})
{{:person.car.brand}} (Model {{:person.car.model|uppercase}})
{{if(person.@@odd)}}this line is odd{{endif}}
{{if(person.@@even)}}this line is even{{endif}}
{{loop(person.colors) as color}}
  {{:color.name}} - {{:color.rgb}}
    color: {{:color.@@index}}
  {{endloop}}

{{endloop}}

** JSON Pseudo Variable **
{{reset(jsonobj)}} {{# ignored }}
{{loop(jsonobj.people) as jarr}}
{{:jarr.@@index}} - {{:jarr.@@odd}} - {{:jarr.@@even}}
{{endloop}}