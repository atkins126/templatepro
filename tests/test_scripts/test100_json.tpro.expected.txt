** JSON **
1) BRUCE BANNER (age 56)
  1° smartphone
  2° dumbphone
Ferrari (Model F40)
this line is odd

  red - ff0000
    color: 1
  blue - 0000ff
    color: 2
  green - 00ff00
    color: 3

2) REED RICHARDS (age 45)
  1° dumbphone
Ferrari (Model 348GTB)

this line is even

3) SCOTT SUMMERS (age 54)
  1° smartphone
  2° desktop
Ford (Model MUSTANG)
this line is odd


4) DANIELE TETI (age 40)
  1° smartphone
  2° dumbphone
Lancia (Model DELTA HF INTEGRALE)

this line is even


** JSON Pseudo Variable **
1 - True - False
2 - False - True
3 - True - False
4 - False - True

<hello>world